File: app/src/main/java/com/example/strawpoll/network/dtos/AnswerDTO.kt
package com.example.strawpoll.network.dtos

import com.example.strawpoll.domain.Answer
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class AnswerDTO(
    val answerId:Int,
    val answerString:String,
    val amountVoted: Int
)

fun List<AnswerDTO>.asDomainModel(): List<Answer> {
    return map {
        Answer(
            id = it.answerId,
            answer = it.answerString,
            amountVoted = it.amountVoted
        )
    }
}

File: StrawpollAPI/PollAPI/DTOs/AnswerDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PollAPI.DTOs
{
    public class AnswerDTO
    {
        public int AnswerId { get; set; }
        public String AnswerString { get; set; }
        public int AmountVoted { get; set; }
    }
}
